e2e/tests/dump-restore-test.rs

```diff
@@ -0,0 +1,304 @@
+        .arg("_timescaledb_catalog")
+        .arg("-N")
+        .arg("_timescaledb_config")
+        .arg("-N")
+        .arg("_timescaledb_internal")
+        .arg("-N")
+        .arg("timescaledb_experimental")
+        .arg("-N")
+        .arg("timescaledb_information")
+        .arg("-f")
+        .arg("/tmp/dump.sql")
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "dumping the database {} via {} user failed: {}",
+        db,
+        username,
+        exit
+    );
+    let path = env::temp_dir().join("dump.sql");
+    if path.exists() {
+        fs::remove_file(path.clone()).expect("failed to remove existing dump file");
+    }
+    let exit = Command::new("docker")
+        .arg("cp")
+        .arg(format!("{}:/tmp/dump.sql", container_id))
+        .arg(path.to_str().unwrap())
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "copying the dump out of the container failed: {}",
+        exit
+    );
+    path
+}
+
+fn restore_db(container_id: &str, db: &str, username: &str, path: PathBuf) {
+    let exit = Command::new("docker")
+        .arg("cp")
+        .arg(path.to_str().unwrap())
+        .arg(format!("{}:/tmp/dump.sql", container_id))
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "copying the dump into the container failed: {}",
+        exit
+    );
+    let exit = Command::new("docker")
+        .arg("exec")
+        .arg(container_id)
+        .arg("psql")
+        .arg("-U")
+        .arg(username)
+        .arg("-d")
+        .arg(db)
+        .arg("--no-password")
+        .arg("--no-psqlrc")
+        .arg("--no-readline")
+        .arg("--echo-all")
+        .arg("-v")
+        .arg("ON_ERROR_STOP=1")
+        .arg("-f")
+        .arg("/tmp/dump.sql")
+        .spawn()
+        .unwrap()
+        .wait()
+        .unwrap();
+    assert!(
+        exit.success(),
+        "restoring database {} via {} user failed: {}",
+        db,
+        username,
+        exit
+    );
+}
+
+#[test]
+fn dump_restore_test() {
+    let docker = clients::Cli::default();
+
+    let container = run_postgres(&docker, "db0", "postgres");
+    let container_id = container.id();
+    psql_script(container_id, "db0", "postgres", "/scripts/init.sql");
+    psql_cmd(
+        container_id,
+        "db0",
+        "postgres",
+        "create user jack; grant prom_admin, postgres to jack;", // todo: jack should not need postgres role
+    );
+    let snapshot0 = snapshot_db(container_id, "db0", "jack");
+    let dump = dump_db(container_id, "db0", "postgres"); // todo: dump using jack user
+    container.stop();
+    container.rm();
+
+    println!("dump file at {}", dump.display());
+
+    let container = run_postgres(&docker, "db1", "postgres");
+    let container_id = container.id();
+    psql_cmd(
+        container_id,
```

Comment:
nit: are we sure it's better to pass in container ids rather than containers into these functions?

Reply:
no, not sure. the functions only need the id, but maybe the typing associated with passing the container itself would be nice? I'll give it a try...

Reply:
okay, that works well enough. changed it.